src/app/iteration/iteration.service.ts

```diff
@@ -80,10 +81,35 @@ export class IterationService {
   }
 
   /**
-   * Get locally saved iterations' reference
+   * Update an existing iteration
+   * @param iteration - Updated iteration
+   * @return updated iteration's reference from the list
    */
-  getIterationList() {
-    return this.iterations;
+  updateIteration(iteration: IterationModel): Promise<IterationModel> {
+    return this.http
+      .patch(iteration.links.self, { headers: this.headers })
+      .toPromise()
+      .then (response => {
+        if (response.status != 200) {
+          throw new Error('API error occured');
```

Comment:
@bartoszmajsak Angular's http handles that smartly. I believe, error from live API goes to catch part by default. This block is only for unit test purpose where we are mocking the response. I see in mock response even if the status is 4xx or 5xx it doesn't go to catch section. 
Though, your point is totally valid. I should just check for error codes instead of != 200. I'll update that.